tests/unit/geom/CoordinateArraySequenceFactoryTest.cpp

```diff
@@ -79,7 +79,7 @@ namespace tut
 			ensure( 0 != col);
 
 			const size_t size0 = 0;
-			CoordinateSequencePtr sequence = factory->create(col);
```

Comment:
I guess you mean `dim3IsM` instead of `dim2isM`? In any event, yes this approach should work equally well and preserve compatibility.

Concerning the `POING(NaN NaN)` aspect, the main issue is probably whether the WKT and WKB parsers are able to correctly handle such values.